### ISSUE TITLE
Intent to implement: using symfony/process

### ISSUE THREAD

**Issue by Adirelle** [Block8/PHPCI#884]
What's the idea ?

[http://symfony.com/doc/current/components/process.html symfony/process] 

Why ?

* It is a well tested component.
* It gracefully handles Windows shenanigans.
* It handles process timeouts (both based on total duration and output idling). Both ca
* It can be used to run PHP in process isolation.

**Comment by dancryer** [comment]
I think that this, combined with #885 - would make for a nice step towards a cleaner command executor.

We'd need to do a thorough review before merging, but would be pleased to see a PR implementing these ideas.

I assume you'll keep the `CommandExecutor` classes, but just make them use `Process` internally?

**Comment by Adirelle** [comment]
Yep, to avoid breaking change, the CommandExecutor would stay as a thin layer between existing code and symfony/process.

**Comment by Adirelle** [comment]
So I started working on this (see there ) and achieved these features:
* catches stdout, stderr and exit code as before,
* "live-streams" stdout and stderr to the loggger.
* implements a per-command and global timeout.

However it the second point gives rise to several questions:
* it doesn't have the exit code while streaming stderr to the logger so it cannot chose not to log it or log it at a level depending on the exit code.
* I've seen they are $logOutput, $verbose and $quiet attributes in Builder and BaseCommandExecutor. I haven't verified how they are used but the plugin yet. Are there some guidelines and/or explanations about what they should do and how to use them ?

Right now, I've implemented the following behaviour but I feel this should be discussed in a more general topic about logging.
* CommandExecutor as new method, execute, that accepts a workingDir, a verbosity level and "out" arguments for exitCode, stdout and stderr feedback.
* the verbosity level accepts four values :
 * STEALTHED: logs neither stdout nor stderr,
 * QUIET: logs stdout and stderr as DEBUG; if the commands exits with a non-zero status, stderr is logged as ERROR again,
 * NORMAL: logs stdout as DEBUG, stderr as NOTICE,
 * VERBOSE: logs stdout as INFO, stderr as NOTICE,
* since Builder is still used by all plugins, it implements the compatibilty layer, converting $quiet, $verbose and $logExecOutpout to a verbosity level and "plublishing" stdout and stderr through getLastOutput() and getLastError().

Any thought ?
